src/toxcore/dht/server/ping.rs

```diff
@@ -0,0 +1,236 @@
+/*
+    Copyright (C) 2013 Tox project All Rights Reserved.
+    Copyright © 2018 Namsoo CHO <nscho66@gmail.com>
+
+    This file is part of Tox.
+
+    Tox is libre software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    Tox is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with Tox.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+
+/*!
+Module for PingRequest
+*/
+
+use std::time::{Duration, Instant};
+use std::mem;
+
+use futures::{future, stream, Stream};
+use parking_lot::RwLock;
+
+use toxcore::dht::packed_node::*;
+use toxcore::dht::kbucket::*;
+use toxcore::dht::server::*;
+use toxcore::dht::dht_node::*;
+use toxcore::io_tokio::IoFuture;
+
+/// Hold data for sending PingRequest
+pub struct Ping {
+    last_time_send_ping: Instant,
+    nodes_to_send_ping: RwLock<Bucket>,
+}
+
+impl Ping {
+    /// new Ping object
+    pub fn new() -> Self {
+        Ping {
+            last_time_send_ping: Instant::now(),
+            nodes_to_send_ping: RwLock::new(Bucket::new(None)),
+        }
+    }
+
+    fn is_bad_node_timed_out(node: &DhtNode, server: &Server) -> bool {
```

Comment:
Maybe move this function to `DhtNode` and just pass `bad_node_timeout` value there?

Reply:
done except some comments which I can't resolve.